examples/rvc-app/rvc-common/include/rvc-service-area-delegate.h

```diff
@@ -47,24 +48,62 @@ class RvcServiceAreaDelegate : public ServiceArea::Delegate
     RvcDevice * mSetSelectedLocationAllowedInstance;
     IsSetSelectedLocationAllowedCallback mSetSelectedLocationAllowedCallback;
 
+    // containers for array attributes
+    std::map<uint32_t, LocationStructureWrapper>       mSupportedLocationsById;     // main storage for supported locations
+    std::vector<LocationStructureWrapper*>             mSupportedLocationsByIndex;  // for fast indexed access - mainly for attribute reads and validations.
```

Comment:
These definitions are private, which in theory gives us nice encapsulation and forces most calls to go through the public ServiceArea Instance, but this is the derived application class, so specific devices can break this if they chose.